include/natalie/random_value.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <utility>
+
+#include "natalie/class_value.hpp"
+#include "natalie/forward.hpp"
+#include "natalie/global_env.hpp"
+#include "natalie/integer_value.hpp"
+#include "natalie/value.hpp"
+#include <random>
+
+namespace Natalie {
+
+using namespace TM;
+
+class RandomValue : public Value {
+public:
+    RandomValue(ClassValue *klass)
+        : Value { Value::Type::Random, klass } { }
+
+    RandomValue()
+        : Value { Value::Type::Random, GlobalEnv::the()->Random() } { }
+
+    virtual void visit_children(Visitor &visitor) override final {
+        Value::visit_children(visitor);
+    }
```

Comment:
You can remove this virtual method since it's just calling the superclass' visit_children.

Reply:
I thought about the general layout of the class. Do you think it makes sense, to store the seed (`m_seed`) as an IntegerValue directly? This way we would not need to create a new Value each time someone calls `Random#seed`. Of course we would need to collect it during GC then...

Reply:
Integers are free if you use `ValuePtr::integer`, since they're encoded in the smart pointer. So either way is fine.